### PR TITLE
change token API requests to satisfy RFC

### PR DESCRIPTION
**Pull request by grounded042**
### Description
We were using JSON to send token requests when the RFC explicitly says to use `application/x-www-form-urlencoded`. This PR fixes that discrepancy.

### References
Fix for #20.

### Testing
Unit tests were added and it was manually tested using Minikube and Auth0.

### CODE REVIEW

File: auth/authorization_tokenretriever.go [hunk 1]
```diff
@@ -76,51 +60,45 @@ func NewTokenRetriever(oidcWellKnownEndpoints OIDCWellKnownEndpoints, authTransp
...
-		bodyReader,
+		strings.NewReader(uv.Encode()),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	request.Header.Add("Content-Length", strconv.Itoa(len(uv.Encode())))
```
Comment (wimdec) [inline auth/authorization_tokenretriever.go]: uv.Encode() done twice.

File: auth/authorization_tokenretriever.go [hunk 2]
```diff
@@ -76,51 +60,45 @@ func NewTokenRetriever(oidcWellKnownEndpoints OIDCWellKnownEndpoints, authTransp
...
```
Comment (wimdec) [inline auth/authorization_tokenretriever.go]: uv.Encode() done twice.

File: auth/authorization_tokenretriever.go (hunk 1 above)
Comment (grounded042) [reply auth/authorization_tokenretriever.go]: Good catch. Fixed.

File: auth/authorization_tokenretriever.go (hunk 2 above)
Comment (grounded042) [reply auth/authorization_tokenretriever.go]: Good catch. Fixed.

**Pull request merged**